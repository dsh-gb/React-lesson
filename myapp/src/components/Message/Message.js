import React from "react"

// компонент Message, вывод текста в формате: автор - текст
function Message(props) {
    const author = props.message.author
    const text = props.message.text
    return (
        <>
            {
                author === 'Me' &&
                < li className="message-me" >
                    <div className="message-author">
                        {author}
                    </div>
                    <div className="message-me-text">
                        {text}
                    </div>
                </li >
            }
            {
                author === 'Bot' &&
                < li className="message-bot" >
                    <div className="message-author">
                        {author}
                    </div>
                    <div className="message-bot-text">
                        {text}
                    </div>
                </li >
            }
        </>
    )
}

export default Message