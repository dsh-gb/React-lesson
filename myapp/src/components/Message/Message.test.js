import { render, screen } from '@testing-library/react'
import Message from './Message'

// тесты для презентационого компонента Message
describe("Component Message", () => {

    // создаем snapshot для компонента Message с данными author="Bot", text="text test" 
    it("Message snapshot author=Bot", () => {
        const component = render(<Message author="Bot" text="text test" />)
        expect(component).toMatchSnapshot()
    })

    // проверяем, если author="Bot" то в тэге <li> есть класс "message-bot"
    it("For author=Bot tag <li> include className='message-bot'", () => {
        render(<Message author="Bot" text="text test" />)
        const liTagMessage = screen.getByText(/Bot/i).parentElement
        expect(liTagMessage).toHaveClass('message-bot')
    })

    // проверяем, если author="Bot" то text находится в div с классом "message-bot-text"
    it("For author=Bot tag <div> with {text} include className='message-bot-text'", () => {
        render(<Message author="Bot" text="text test" />)
        const divTagMessage = screen.getByText(/text test/)
        expect(divTagMessage).toHaveClass('message-bot-text')
    })
})