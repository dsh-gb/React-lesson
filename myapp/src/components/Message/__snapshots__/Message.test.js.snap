// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Message Message snapshot author=Bot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="message-bot"
      >
        <div
          class="message-author"
        >
          Bot
        </div>
        <div
          class="message-bot-text"
        >
          text test
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="message-bot"
    >
      <div
        class="message-author"
      >
        Bot
      </div>
      <div
        class="message-bot-text"
      >
        text test
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
